ddpui/core/orchestrate/pipeline_service.py

```
@@ -0,0 +1,789 @@
+                total_run_time=-1,
+                status="Scheduled",
+                state_name="Scheduled",
+                retries=0,
+                orguser=orguser,
+            )
+        except Exception as error:
+            for task_lock in locks:
+                logger.info("deleting TaskLock %s", task_lock.orgtask.task.slug)
+                task_lock.delete()
+            logger.exception(error)
+            raise PipelineServiceError(f"Failed to start a run {str(error)}")
```

Ruff (0.15.10)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)